### PR TITLE
Fix for gemma3 issue

### PR DESCRIPTION
**Pull request by quic-hemagnih**
**Issue : When compiled w/ sub-function following error was thrown,**
Error: [Operator-'Range_343', opset_version-17, ir_version-8] : Range: 'limit' input must be a constant tensor (found node "onnx::Range_1524").

**Fix :**
`attention_mask` is built once at the model level (see ``QEffGemma3TextModel.forward``) and passed in per layer-type. Building it here would emit a ``torch.arange`` inside the decoder-layer subfunction during ONNX subfunction export, which the AIC compiler rejects because the ``Range`` ``limit`` input becomes a function parameter rather than a constant.
In QEffGemma3TextModel.forward, the else branch for QEffSlidingWindowCache now builds the same two masks the old decoder-layer code built, but at the model level:
      - "sliding_attention" → _create_causal_mask(target_length=sliding_window_len, sliding_window=sliding_window_len)
      - "full_attention" → _create_causal_mask(target_length=key_cache[pattern-1].shape[-2])
The torch.arange inside _create_causal_mask now executes in the model-level forward, outside any subfunction boundary.

Also added spli-model-io flag in the example script to fix output issue.

**Testing :**

1. Was able to compile sucessfully w/ sub functions.
2. Test results w/o sub functions (legacy code), using example script,
["Here's a detailed description of the image:\n\n**Overall Impression:**\n\nThe image features a remarkably realistic-looking cat dressed in formal attire, creating a whimsical and charming scene. The cat appears to be the central focus, set within a lush garden environment.\n\n**The Cat:**\n\n*   **Breed/Appearance:** The cat is a light orange tabby with white markings on its chest, face, and paws. It has large, expressive, bright blue eyes that are a striking feature.\n*   **Attire:** The cat is impeccably dressed in a miniature suit. It wears a blue jacket with gold buttons, a white shirt with a brown bow tie, and a yellow vest. The suit is tailored to fit the cat's form, adding to the surreal and delightful quality of the image.\n*   **Pose:** The cat is sitting upright, looking slightly upwards and to the right with a thoughtful or curious expression. Its posture is dignified and almost human-like.\n\n**Background and Setting:**\n\n*   **Garden:** The cat is situated in a garden setting. There's a stone pathway leading into the background, and lush greenery surrounds the cat.\n*   **Flowers:** White flowers are prominently featured in the foreground, adding a touch of freshness and vibrancy to the scene.\n*   **Building:** In the background, a quaint, thatched-roof cottage is partially visible, suggesting a rural or countryside location.\n*   **Lighting:** The lighting appears to be soft and natural, with a slightly diffused quality. This enhances the realism and creates a pleasant atmosphere.\n\n**Overall Style and Tone:**\n\n*   **Realism:** The image is rendered with a high degree of realism, making the cat and its surroundings appear almost tangible.\n*   **Whimsical:** Despite the realism, the concept of a cat dressed in a suit is inherently whimsical and playful.\n*   **Charming:** The overall tone is charming and inviting, evoking a sense of wonder and delight.\n\nIn essence, the image is a beautifully crafted and imaginative portrayal of a cat in a human-like role, set within a picturesque garden environment."]
Average Prefill time a.k.a TTFT is= 1.59 sec        
Decode is= 3.89 token/sec        
Total is= 3.88 token/sec        
Total (E2E) inference time is= 515.3 sec

3. W/ subfunctions we would need a separate NPI file.

### CODE REVIEW

File: QEfficient/transformers/models/gemma3/modeling_gemma3.py [hunk 1]
```diff
@@ -446,7 +437,23 @@ def forward(
                 ),
             }
         else:
-            _causal_mask_mapping = None
+            # QEffSlidingWindowCache path (ONNX export / on-device runtime).
+            # Replicate the per-layer-type mask logic that previously lived inside
+            # QEffGemma3DecoderLayer.forward, but build it once here so that the
+            # torch.arange inside _create_causal_mask stays outside any subfunction.
+            sliding_window_len = past_key_values.sliding_window_len
+            full_ctx_len = past_key_values.key_cache[self.config._sliding_window_pattern - 1].shape[-2]
```
Comment (vbaddi) [inline QEfficient/transformers/models/gemma3/modeling_gemma3.py:445]: This can be move to the TextModel path too?

**Review by vbaddi — CHANGES_REQUESTED** [review]

File: tests/diffusers/test_wan_i2v.py [hunk 2]
```diff
@@ -37,6 +38,20 @@
 TEST_SEED = 42
 
 
+def _cleanup_pipeline_qpc_sessions(pipeline) -> None:
```
Comment (vbaddi) [inline tests/diffusers/test_wan_i2v.py]: We had to do this manually? shouldn't deactivate handle this?

Comment (quic-hemagnih) [reply tests/diffusers/test_wan_i2v.py]: Ideally deactivate should handle this, but I am explicity cleaning off all the contexts, want to run a full CI to check and see the results.

File: QEfficient/transformers/models/gemma3/modeling_gemma3.py (hunk 1 above)
Comment (quic-hemagnih) [reply QEfficient/transformers/models/gemma3/modeling_gemma3.py:445]: Its in the QEffGemma3TextModel class

File: tests/unit_test/models/test_model_quickcheck.py [hunk 3]
```diff
@@ -578,6 +578,42 @@ def test_causal_subfunction_export_smoke(tmp_path):
     assert not any("QEffGPT2Block" in name for name in without_names)
 
 
+@pytest.mark.llm_model
+def test_gemma3_vlm_export_parity_with_and_without_subfunctions(tmp_path):
```
Comment (vbaddi) [inline tests/unit_test/models/test_model_quickcheck.py:587]: nit: what is this test checking for? Only IO comparision? if we want to check for w/subf, let's check its modules via model.functions()?

Comment (quic-hemagnih) [reply tests/unit_test/models/test_model_quickcheck.py:587]: Incorporated the review comments

**Comment by quic-hemagnih** [comment]
["Here's a detailed description of the image:\n\n**Overall Impression:**\n\nThe image features a remarkably realistic-looking cat dressed in formal attire, creating a whimsical and charming scene. The cat appears to be the central focus, set within a lush garden environment.\n\n**The Cat:**\n\n*   **Breed/Appearance:** The cat is a light orange tabby with white markings on its chest, face, and paws. It has large, expressive, bright blue eyes that are a striking feature.\n*   **Attire:** The cat is impeccably dressed in a miniature suit. It wears a blue jacket with gold buttons, a white shirt with a brown bow tie, and a yellow vest. The suit is tailored to fit the cat's form, adding to the surreal and delightful quality of the image.\n*   **Pose:** The cat is sitting upright, looking slightly upwards and to the right with a thoughtful or curious expression. Its posture is dignified and almost human-like.\n\n**Background and Setting:**\n\n*   **Garden:** The cat is situated in a garden setting. There's a stone pathway leading into the background, and lush greenery surrounds the cat.\n*   **Flowers:** White flowers are prominently featured in the foreground, adding a touch of freshness and vibrancy to the scene.\n*   **Building:** In the background, a quaint, thatched-roof cottage is partially visible, suggesting a rural or countryside location.\n*   **Lighting:** The lighting appears to be soft and natural, with a slightly diffused quality. This enhances the realism and creates a pleasant atmosphere.\n\n**Overall Style and Tone:**\n\n*   **Realism:** The image is rendered with a high degree of realism, making the cat and its surroundings appear almost tangible.\n*   **Whimsical:** Despite the realism, the concept of a cat dressed in a suit is inherently whimsical and playful.\n*   **Charming:** The overall tone is charming and inviting, evoking a sense of wonder and delight.\n\nIn essence, the image is a beautifully crafted and imaginative portrayal of a cat in a human-like role, set within a picturesque garden environment."]
Average Prefill time a.k.a TTFT is= 1.59 sec        
Decode is= 6.77 token/sec        
Total is= 6.73 token/sec        
Total (E2E) inference time is= 296.83 sec

**Review by tchawada — APPROVED** [review]

**Comment by quic-hemagnih** [comment]
CI is passed, merging this PR in mainline. 
https://qraniumci.qualcomm.com/blue/organizations/jenkins/quic_efficient-transformer_public/detail/PR-1101/18/pipeline
